doc: Adds known limitations document (#12)
canonical/chaos-engineering: docs/reference/known_limitations.md

# Known limitations

> Here we list the known limitations of the chaos engineering platform in its current state.

Reviewer: ```suggestion
Here we list the known limitations of the Charmed Chaos Engineering platform in its current state.
```

Reviewer: That's to keep consistency across the docs.

These are wide-scoped missing features (or products) that we would like to work on eventually.

For a list of known issues, bugs and feature requests that are currently being worked on or being tracked, refer to these issue trackers:

- https://github.com/canonical/chaos-engineering/issues
- https://github.com/canonical/litmus-operators/issues
- https://github.com/litmuschaos/litmus/issues

> ## Chaos testing for machine clouds.

Reviewer: ```suggestion
## Chaos testing for machine clouds
```

> At the moment, the chaos engineering platform focuses on chaos-testing for k8s clouds.

Reviewer: ```suggestion
At the moment, the Charmed Chaos Engineering platform focuses on chaos-testing for k8s environments.
```

We are currently not looking into charmed VM cloud chaos testing; but that is definitely an interesting avenue for future development. Thoughts on the matter? Get in touch!

> ## Charmed Litmus chaos execution plane.

Reviewer: ```suggestion
## Charmed Litmus execution plane
```

> At the moment, the `litmus-operators` project only wraps the Chaos Control plane. Charming the Execution plane is something we'll be looking into in the near future, so stay tuned!

Reviewer: ```suggestion
At the moment, the Charmed Litmus project only wraps the Control plane. Charming the Execution plane is something we'll be looking into in the near future, so stay tuned!
```

> ## TLS integration for the Litmus control plane.

Reviewer: ```suggestion
## TLS integration for the Litmus Control plane
```

Due to [an upstream bug](https://github.com/litmuschaos/litmus/issues/3136), at the moment the Litmus Control plane cannot talk over TLS to mongodb.
The `mongodb-k8s` charm's quite tolerant TLS integration approach means that even if MongoDB has a tls-certificates relation, it will not reject the Litmus components' traffic, but the traffic will nonetheless be unencrypted. So beware of this issue, and keep an eye on the upstream issue for when a fix will eventually be shipped.

> ## Admin password syncing for the Litmus Control plane.

Reviewer: ```suggestion
## Admin password syncing for the Litmus Control plane
```

At the moment, Litmus doesn't wipe the MongoDB tables upon teardown, which means that if you change the admin password and forget it, the only way to 'reset' it is manually deleting the tables in MongoDB or redeploying the `mongodb-k8s` charm altogether.

> So beware that even if you redeploy the control plane, the user settings will be persisted in the database.

Reviewer: ```suggestion
So beware that even if you redeploy the Control plane, the user settings will be persisted in the database.
```
